Lib/multiprocessing/managers.py

```diff
@@ -159,6 +159,10 @@ def __init__(self, registry, address, authkey, serializer):
         self.id_to_local_proxy_obj = {}
         self.mutex = threading.Lock()
 
+    def _track_parent(self):
+        process.parent_process().join()
```

Comment:
What if `process.parent_process()` returns None? For example if the Server was launched in an independent Python instance?

Reply:
Did not know this was a supported use-case. But we can add a guard.

Reply:
However, adding a guard will simply discard the case where `parent_process() is None`, which is not the best solution. Yet, I feel that tracking the usage of manager-allocated resources by other processes than the manager's direct parent is tricky, especially for `SharedMemoryManager` (see more detailed explanation here https://github.com/python/cpython/pull/13247#issuecomment-493694766).

Do we want to make this feature more robust by trying to make the manager wait on any process that tries to connect to it?

Reply:
I'm afraid I'm not following you here. We're simply trying to fix the case where the manager was launched by its parent process, no? (that's what the issue title says)

The case where the manager is autonomous looks pretty much unfixable on its own.

Reply:
> We're simply trying to fix the case where the manager was launched by its parent process

Ok, lets stick to this in this PR.

> The case where the manager is autonomous looks pretty much unfixable on its own.

I agree, but there are in-betweens: if a manager is created by a process (its parent), and then connected to by other processes, then should the manager just watch its parent? Technically, if other processes are also connected to it, it probably should not shut down unless ALL processes connected to it (and not only its parent) terminated.

Reply:
Hmm... So, conversely, if the manager loses all its connections _and_ was created by a parent process, then it should die?

Perhaps that should be the right heuristic?

Reply:
Yes, I think this is already a more robust solution. However, as opposed to with `SyncManager`, manipulating/modifying `SharedMemoryManager` delivered memory segments does not necessarily trigger communication with the manager server process. This is because `SharedMemoryManager` delivers `SharedMemory` objects, and not proxies. Thus, keeping track of the number of connected clients does not equate keep track of the processes manipulating those objects. But maybe this approach is still good enough for now.

Also, the recent `os.memfd_create` mentioned in [this python issue](https://bugs.python.org/issue26836) looks interesting, but I need to look more into it.